src/arena/game_state.rs

```diff
@@ -331,6 +332,48 @@ impl GameState {
                 }
             });
 
+        let mut times_acted = vec![0; num_players];
+
+        let midgame = round != Round::Starting;
+
+        if midgame {
+            if ante > 0.0 {
+                for action_count in times_acted.iter_mut() {
+                    *action_count += 1; // Everyone antes
+                }
+            }
+
+            if small_blind > 0.0 {
+                if num_players > 2 { // Full-ring
+                    let left_of_dealer_idx = (dealer_idx + 1) % num_players;
+
+                    if let Some(action_count) = times_acted.get_mut(left_of_dealer_idx) {
+                        *action_count += 1;
+                    }
+                } else if num_players == 2 { // Heads-up
+                    if let Some(action_count) = times_acted.get_mut(dealer_idx) {
+                        *action_count += 1;
+                    }
+                }
+            }
+
+            if big_blind > 0.0 {
+                if num_players > 2 { // Full-ring
+                    let two_left_of_dealer_idx = (dealer_idx + 2) % num_players;
+
+                    if let Some(action_count) = times_acted.get_mut(two_left_of_dealer_idx) {
+                        *action_count += 1;
+                    }
+                } else if num_players == 2 { // Heads-up
+                    let other_player_idx = (dealer_idx + 1) % num_players;
+
+                    if let Some(action_count) = times_acted.get_mut(other_player_idx) {
+                        *action_count += 1;
+                    }
+                }
+            }
+        }
+
```

Comment:
After thinking about it more, I didn't like the idea of putting this in gamestate. I really wanted this to be outside. So I created a historian that does this.

Can you please remove this part and rebase on master?